code-challenges.test.js

```diff
@@ -14,33 +14,114 @@
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+describe("remainder", () => 
+{
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+  const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+
+it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+  expect(remainder(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+  expect(remainder(hodgepodge2)).toEqual([ 2, 1, -1 ])
+})
+})
+
+/*
+ FAIL  ./code-challenges.test.js
+  createSentence
+    ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (2 ms)
+  remainder
+    ✕ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+  ● remainder › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+    ReferenceError: remainder is not defined
+*/
 
 // b) Create the function that makes the test pass.
 
+const remainder = (array) => {
+  // iterates over the array and filter out non-numeric values
+  const numbers = array.filter(element => typeof element === 'number')
+  // iterates over the numbers array and computes the remainder of each number when divided by 3
+  const remainders = numbers.map(number => number % 3)
+  // returns the array of remainders
+  return remainders
+}
+
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+describe("sumCubedNumbers", () => 
+{
+  const cubeAndSum1 = [2, 3, 4]
+  const cubeAndSum2 = [0, 5, 10]
+
+it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+  expect(sumCubedNumbers(cubeAndSum1)).toEqual(99)
+  expect(sumCubedNumbers(cubeAndSum2)).toEqual(1125)
+})
+})
+
+/*
+ FAIL  ./code-challenges.test.js
+  createSentence
+    ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (2 ms)
+  remainder
+    ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3. (1 ms)
+  sumOfNumbersCubed
+    ✕ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+  ● sumOfNumbersCubed › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+    ReferenceError: sumOfNumbersCubed is not defined
+*/
 
 // b) Create the function that makes the test pass.
+
+const sumCubedNumbers = numbers => numbers.reduce((sum, number) => sum + Math.pow(number, 3), 0);
+// reduce calculates the sum of the cubed numbers. 
```

Comment:
That's what it's doing overall, but not exactly what reduce does on it's own. This pseudo code does not fully explain the logic here.